posts/2022/02/making-gpus-accessible-to-pydata-ecosystem-via-array-api.md

```diff
@@ -0,0 +1,481 @@
+decade a plethora of array libraries have evolved, which more or less derives
+from NumPy’s API and it’s API is not very well defined. This is due to the fact
+that NumPy is a CPU-only library and today these dependent libraries are built
+for a variety of exotic hardware. As a consequence the APIs of these libraries
+have diverged a lot, So much so that it’s quite difficult to write code that
+works with multiple (or all) of these libraries. The Array API standard aims
+to address this issue, by specifying an API for the most common ways arrays
+are constructed and used. At the time of writing this, NumPy (>=1.22.0) and
+CuPy (>=10.0.0) have adopted the Array API Standard.
+
+
+##### Note about using Array API Standard
+
+The definition of Array API Standard and it’s usage may evolve over time and
+this blog post is just an example of how we can use it to make GPUs accessible
+to the users of libraries like scipy and scikits and it doesn’t not necessarily
+define the best practices around adopting the standard. The best practices for
+adopting Array API is still a work very much in progress and will evolve drastically.
+
+
+## Demonstration Example
+
+The example we have chosen to demonstrate adopting the Array API standard to make
+GPUs accessible in SciPy and scikits is the one taken from scikit-learn’s
+documentation: Segmenting the picture of greek coins in regions. This example
+uses Spectral clustering on a graph created from voxel-to-voxel difference on
+an image to break this image into multiple partly-homogeneous regions (or say clusters).
+In essence spectral clustering is about finding the clusters in a graph i.e.
+finding the mostly connected subgraph and thereby identifying the clusters alternatively
+it can be described as finding subgraphs having maximum number of within cluster connections
+and minimum number of between cluster connections.
+
+### Dataset
+
+<p align="center">
+    <img
+     alt="Greek coins from Pompeii"
+     src="/images/2022/02/greek_coins_original.jpeg">
+</p>
+<p align="center">
+<i>Greek coins from Pompeii ("British Museum, London, England")</i>
+</p>
+
+The dataset we are using for this demonstration is of greek coins from Pompeii
+(from skimage.data.coins module). This is a data of image of several coins outlined
+against a gray background.
+
+We convert the data of the coins into a graph object and then apply spectral
+clustering to it for segmenting coins.
+
+Source: [https://www.brooklynmuseum.org/opencollection/archives/image/51611](https://www.brooklynmuseum.org/opencollection/archives/image/51611)
+
+## Process
+
+#### `get_namespace` - getting the array API module
+
+The [NEP 47](https://numpy.org/neps/nep-0047-array-api-standard.html) (Adopting
+the array API standard) outlines a basic workflow for adopting array API
+standards for array provider and consumer libraries. Now we will describe
+in detail about how we went through the process of implementing this.
```

Comment:
@aktech should this sentence say _"of implementing this in forks of SciPy, scikit-learn and scikit-image."_?